src/server/mod.rs

```diff
@@ -34,4 +39,41 @@ pub use self::node::{create_raft_storage, Node};
 pub use self::resolve::{PdStoreAddrResolver, StoreAddrResolver};
 pub use self::raft_client::RaftClient;
 
-pub type OnResponse = Box<FnBox(Response) + Send>;
+pub type CopStream<T> = Box<Stream<Item = T, Error = mpsc::SendError<T>> + Send>;
+
+pub enum OnResponse<T> {
+    Unary(Box<FnBox(T) + Send>),
+    Streaming(Box<FnBox(CopStream<T>, Option<CpuPool>) + Send>),
```

Comment:
Is the `Box` necessary?

Reply:
I think so because we have a test case using it. See [here](https://github.com/pingcap/tikv/pull/2478/files#diff-c1fcd538536072b17fa388c7722dbc17R1477).